src/hooks/useInterval.js

```diff
@@ -0,0 +1,26 @@
+import React, {useEffect, useRef } from 'react'
+
+function useInterval(callback, delay) {
+  const savedCallback = useRef()
+  let firstTime = true
+
+  // Remember the latest callback.
+  useEffect(() => {
+    savedCallback.current = callback
+  }, [callback])
+
+  // Set up the interval.
+  useEffect(() => {
+    function tick() {
+      if (!firstTime)
+        savedCallback.current()
+        firstTime = false
```

Comment:
Codacy found an issue: [Assignments to the 'firstTime' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5837031)